src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -1013,10 +1013,72 @@ zocl_load_aie_only_pdi(struct drm_zocl_dev *zdev, struct axlf *axlf,
 }
 
 int
-zocl_pl_only_reset(struct drm_zocl_dev *zdev, const char *buf, size_t count)
+zocl_reset(struct drm_zocl_dev *zdev, const char *buf, size_t count)
 {
-	/* TODO */
-	DRM_ERROR("%s PL Reset support doesn't added yet", __func__);
+	struct drm_zocl_slot *z_slot = NULL;
+	struct kds_sched *kds = &zdev->kds;
+	struct kds_client *client = NULL;
+	struct kds_client *tmp_client = NULL;
+	void __iomem *map = NULL;
+	int ret = 0;
+	int i = 0;
+
+	if (strcmp(zdev->zdev_data_info->fpga_driver_name, "versal_fpga"))
+		return count;
+
+	/* Find out number of active client and send a signal to it. */
+	list_for_each_entry_safe(client, tmp_client, &kds->clients, link) {
+		ret = kill_pid(client->pid, SIGTERM, 1);
+		if (ret) {
+			DRM_WARN("Failed to terminate Client pid %d."
+			       " Performing SIGKILL.\n", pid_nr(client->pid));
+			kill_pid(client->pid, SIGKILL, 1);
+		}
+	}
+
```

Comment:
It is better that acquired kds->lock to protect this loop.

Reply:
Will protect this. Thanks for suggestion.